src/containers/MemoryGame/index.jsx

```diff
@@ -1,12 +1,44 @@
-import React from "react";
+import React, { useState } from "react";
 import "./index.css";
-import MemoryGameCards from "../../components/memorygame/MemoryGameCards/index";
+import { Col, Row } from "antd";
+import { useTranslation } from "react-i18next";
+import Confetti from "react-dom-confetti";
+import confettiConfig from "../../components/memorygame/confettiConfig";
+import MemoryGameFactList from "../../components/memorygame/MemoryGameFactList";
+import MemoryGameBoard from "../../components/memorygame/MemoryGameBoard";
 
 const MemoryGame = () => {
+  const [t] = useTranslation();
+  const [isGameCompleted, setIsGameCompleted] = useState(false);
+  // finishedCards is being used to display environmental facts.
+  // It is an array of object(states).Each object is a successfully matched card state.
+  // When a match is made successfully only state of first card gets pushed inside finishedCards.
+  // Because matched cards have same state. Otherwise same fact would be displayed twice.
+  const [finishedCards, setFinishedCards] = useState([]);
+
   return (
-    <div className="memoryGameContainer">
-      <MemoryGameCards />
-    </div>
+    <Row justify="center" className="memoryGameContainer">
+      {/* Confetti displays confetti effect to congratulate user when game is completed */}
+      {/* Check https://github.com/daniel-lundin/react-dom-confetti for 'react-dom-confetti' */}
+      <Confetti active={isGameCompleted} config={confettiConfig} />
+      <Col xs={22} md={20} xl={17}>
+        <h1 className="memoryGameContainer__title">{t("memoryGame.title")}</h1>
+        <h3 className="memoryGameContainer__description">
+          {t("memoryGame.header")}
+        </h3>
+        <Row justify="center">
+          <MemoryGameBoard
+            setFinishedCards={setFinishedCards}
+            finishedCards={finishedCards}
+            setIsGameCompleted={setIsGameCompleted}
+          />
+        </Row>
+      </Col>
+      <Col xs={20}>
+        <Confetti active={isGameCompleted} config={confettiConfig} />
+        <MemoryGameFactList facts={finishedCards} />
```

Comment:
related to the last comment, as a general rule, components should only know what they need to. if i had to write tests for MemoryGameFactList, i'd have to write a bunch of things about cards being flipped and so on. and also i'd have to remember that it should only take de-deduplicated card states

this would be an example of an unclean interface. components should be "dumb" and well-defined. memorygamefactlist should be "dumb" in the sense that it should say: "i don't know anything about card states. i don't know anything about flipping. all i know is that you're gonna give me some facts and i'm going to show them."

also the fact that it has a prop called `facts` that takes card states already seems wrong (the interface is good in spirit but facts !== cardState in terms of "types")

Reply:
Understood.